client-lib-tests/src/test/java/com/ibm/ws/repository/resources/internal/test/EsaResourceImplTest.java

```diff
@@ -0,0 +1,143 @@
+        assertEquals("Wrong number of features", 2, features2.size());
+        assertTrue("Features doesn't contain the expected value", features2.contains("bar"));
+
+        Map<String, Collection<String>> fwt2 = toTest.getRequireFeatureWithTolerates();
+        assertEquals("Wrong number of features", 2, fwt2.size());
+        Collection<String> tolerates2 = fwt2.get("bar");
+        assertNotNull("The feature bar didn't appear in the collection", tolerates2);
+        assertTrue("The tolerate for feature bar didn't appear in the tolerates collection",
+                   tolerates2.contains("tolerated"));
+
+        // Test setting an old style feature, should overwrite the previous data.
+        toTest.setRequireFeature(Collections.singleton("set_foo_feature"));
+        Collection<String> features3 = toTest.getRequireFeature();
+        assertEquals("Wrong number of features", 1, features3.size());
+        assertTrue("features3 doesn't contain the 'set_foo_feature' feature", features3.contains("set_foo_feature"));
+
+        Map<String, Collection<String>> fwt3 = toTest.getRequireFeatureWithTolerates();
+        assertEquals("Wrong number of features", 1, fwt3.size());
+        Collection<String> tolerates3 = fwt3.get("set_foo_feature");
+        assertNotNull("The feature set_foo_feature didn't appear in the collection", tolerates3);
+        assertEquals("There should have been no tolerates", 0, tolerates3.size());
+
+        // Test setting a new style feature with tolerates info
+        Collection<String> tolerated4 = Collections.singleton("tolerated4");
+        Map<String, Collection<String>> something4 = Collections.singletonMap("feature4", tolerated4);
+        toTest.setRequireFeatureWithTolerates(something4);
+        Collection<String> features4 = toTest.getRequireFeature();
+        assertEquals("Wrong number of features", 1, features4.size());
+        assertTrue("features4 doesn't contain the expected 'feature4' feature", features4.contains("feature4"));
+
+        Map<String, Collection<String>> fwt4 = toTest.getRequireFeatureWithTolerates();
+        assertEquals("The wrong number of features", 1, fwt4.size());
+        Collection<String> tolerates4 = fwt4.get("feature4");
+        assertNotNull("There should have been some tolerates", tolerates4);
+        assertTrue("tolerates4 didn't contain the 'tolerated4' feature", tolerates4.contains("tolerated4"));
+
+    }
+
+    @Test
+    public void testCopyRequireFeatureField() {
+        Asset asset = new Asset();
+        WlpInformation wlp = new WlpInformation();
+        ArrayList<String> required = new ArrayList<String>();
+        required.add("la di dah");
+        wlp.setRequireFeature(required);
+        asset.setWlpInformation(wlp);
+        EsaResourceImpl esa = new EsaResourceImpl(null, asset);
+        Collection<String> features = esa.getRequireFeature();
+        assertEquals("wrong size", 1, features.size());
+
```

Comment:
Would it be valid to test here that `getRequireFeatureWithTolerates()` returns the one feature, before adding an additional feature?

Would it also be valid to check that doing so does not set the field in the `WlpInformation` object?